client/src/hooks/useMyDataset.js

```diff
@@ -360,6 +360,12 @@ export const useMyDataset = () => {
     const datasetDataCopy = structuredClone(myDataset.data) || {}
 
     datasetDataCopy[project.scpca_id] = newProjectData
+    if (
+      newProjectData.SINGLE_CELL?.length === 0 &&
+      newProjectData.SPATIAL?.length === 0
+    ) {
+      delete datasetDataCopy[project.scpca_id]
+    }
 
```

Comment:
Could you please assign `newProjectData` to the project data only if it contains samples, instead of doing it upfront? This will help optimize assignments and minimize steps 👍

```suggestion 

     // Remove the project if it has no samples
    if (
      newProjectData.SINGLE_CELL?.length === 0 &&
      newProjectData.SPATIAL?.length === 0
    ) {
      delete datasetDataCopy[project.scpca_id]
    } else {
      datasetDataCopy[project.scpca_id] = newProjectData
    }

```